mesh/mesh.go

```diff
@@ -382,6 +382,8 @@ func (msh *Mesh) applyResults(ctx context.Context, results []result.Layer) error
 				}
 			}
 			if err := msh.executor.Execute(ctx, layer.Layer, block); err != nil {
+				// TODO(mafa): sometimes this fails because the block executed references a tx that is not in the DB
+				// maybe in that case the node should try to fetch the missing txs and retry executing the block?
```

Comment:
Add issue, systest run with that problem: https://grafana.spacemesh.dev/goto/r3o5AFKNR?orgId=1